### ISSUE TITLE
[Bug Report] TransformerBridge.run_with_cache does not support backward gradient caching (incl_bwd) unlike HookedRootModule.run_with_cache

### ISSUE THREAD

**Issue by BashCache** [TransformerLensOrg/TransformerLens#1680]
**Describe the bug**
I am using TransformerBridge and noticed a difference between the behavior/API of run_with_cache() for HookedRootModule and TransformerBridge.

The HookedRootModule.run_with_cache() API supports:
incl_bwd: bool = False

When incl_bwd=True, it creates backward caching hooks and stores gradients using the _grad suffix. For example:

> blocks.0.attn.hook_pattern
> blocks.0.attn.hook_pattern_grad

However, the TransformerBridge.run_with_cache() implementation currently appears to only install forward caching hooks.

Its implementation creates hooks using `hp.add_hook(make_cache_hook(name))`

and the cache hook stores `cache[name] = tensor.detach()`

The implementation does not appear to:
* accept incl_bwd as an explicit argument,
* create backward hooks,
* call output.backward(), or
* store gradients with a _grad suffix.

The current TransformerBridge.run_with_cache() signature is:
```
def run_with_cache(
    self,
    input,
    return_cache_object=True,
    remove_batch_dim=False,
    names_filter=None,
    stop_at_layer=None,
    **kwargs,
)
```

while the HookedRootModule.run_with_cache() API includes:
incl_bwd: bool = False.

Backward hooks work correctly when explicitly using run_with_hooks() with the model_bridge module

**Code example**
```
import torch
from transformer_lens import TransformerBridge

model.train()

tokens = model.to_tokens(
    "The capital of France is Paris."
)

loss, cache = model.run_with_cache(
    tokens,
    return_type="loss",
    names_filter=[
        "blocks.0.attn.hook_pattern",
        "blocks.8.attn.hook_z",
    ],
    incl_bwd=True,
)

print("Loss:", loss)

for key in cache.keys():
    print(key)
```

Only forward activations are present:

blocks.0.attn.hook_pattern
blocks.8.attn.o.hook_in
blocks.8.attn.hook_z

**System Info**
Describe the characteristic of your environment:
 * Describe how `transformer_lens` was installed (pip)
 * What OS are you using? (Linux) 
 * Python version (Python3.10)

**Additional context**
Is the lack of backward gradient caching in TransformerBridge.run_with_cache() intentional?

If so, it might be helpful to document that TransformerBridge.run_with_cache() currently only supports forward activation caching and does not support incl_bwd.

If not, it would be useful for TransformerBridge.run_with_cache() to support an explicit:

incl_bwd: bool = False

argument, similar to HookedRootModule.run_with_cache(), and cache backward gradients with the _grad suffix.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)

**Comment by jlarson4** [comment]
Thanks for the report on this @BashCache, this slipped by me. All the tests designed to cache it were paper thin. Cleaning it up now and should have a 3.7.3 release later today with the fix

**Comment by jlarson4** [comment]
@BashCache The release went out just now, this should be resolve if it is relevant to work you are doing!

**Comment by BashCache** [comment]
That was real quick! Thanks a ton!

**Comment by jlarson4** [comment]
You're welcome @BashCache! If I know someone is using a feature and they request a fix, I do my best to get it to them ASAP. Always happy to support the community.
